common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -510,6 +510,67 @@ public UTF8String trim() {
     }
   }
 
+  /**
+   * Removes all specified trim character string either from the beginning or the ending of a string
+   * @param trimString the trim character string
+   */
+  public UTF8String trim(UTF8String trimString) {
+    // this method do the trimLeft first, then trimRight
+    int s = 0; // the searching byte position of the input string
+    int i = 0; // the first beginning byte position of a non-matching character
+    int e = 0; // the last byte position
+    int numChars = 0; // number of characters from the input string
+    int[] stringCharLen = new int[numBytes]; // array of character length for the input string
+    int[] stringCharPos = new int[numBytes]; // array of the first byte position for each character in the input string
+    int searchCharBytes;
+
+    while (s < this.numBytes) {
+      UTF8String searchChar = copyUTF8String(s, s + numBytesForFirstByte(this.getByte(s)) - 1);
+      searchCharBytes = searchChar.numBytes;
+      // try to find the matching for the searchChar in the trimString set
+      if (trimString.find(searchChar, 0) >= 0) {
```

Comment:
From the Oracle doc: "the function searches for each character in trim_source, in turn, removing characters from trim_source until it encounters a character in trim_source that is not in trim-characters."